falcon/testing/base.py

```diff
@@ -47,13 +47,19 @@ class TestBase(unittest.TestCase):
             can use to add a route to the API.
     """
 
-    def setUp(self):
+    def setUp(self, api=None, srmock=None):
         """Initializer, unittest-style"""
 
         super(TestBase, self).setUp()
         self._id = itertools.count(0)
-        self.api = falcon.API()
-        self.srmock = StartResponseMock()
+        if api is not None:
+            self.api = api
+        else:
+            self.api = falcon.API()
+        if srmock is not None:
+            self.srmock = srmock
+        else:
+            self.srmock = StartResponseMock()
         self.test_route = '/{0}'.format(next(self._id))
```

Comment:
Nitpick: Add a blank line above L63 to show it is not related to the manipulation of srmock in L59-62